src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -178,6 +178,20 @@ public static bool ReportWithoutCrashUnlessCanceledAndPropagate(Exception except
             return false;
         }
 
+        /// <summary>
+        /// Suppress finalization of an instance when an exception is thrown from the constructor.
+        /// </summary>
+        /// <typeparam name="T">The type being created.</typeparam>
+        /// <param name="instance">The partially-constructed instance for which finalization should be suppressed.</param>
+        /// <returns>Always returns <see langword="false"/> to propagate the exception.</returns>
+        [DebuggerHidden]
+        public static bool SuppressFinalizeForFailedConstruction<T>(T instance)
```

Comment:
Unused?

Reply:
Currently yes. I'm still investigating patterns for some other cases and this seems like a well-defined operation should it be needed.

Reply:
I would rather not add dead code. Let's add it when it's needed.

Reply:
Removed